zk-kyc/zkvm/host/src/main.rs

```diff
@@ -0,0 +1,424 @@
+use anyhow::{anyhow, Context, Result};
+use bonsai_sdk::alpha::responses::SnarkReceipt;
+use clap::Parser;
+use futures::StreamExt;
+use host::seal_serializer::SnarkReceiptWrapper;
+use log::{error, info};
+use methods::JWT_VALIDATE_ELF;
+use oidc_validator::IdentityProvider;
+use pretty_env_logger;
+use risc0_zkvm::{
+    default_executor, serde::to_vec, ExecutorEnv, MemoryImage, Program, GUEST_MAX_MEM, PAGE_SIZE,
+    VERSION,
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    sync::{atomic::AtomicUsize, Arc},
+};
+use tokio::sync::{mpsc, RwLock};
+use tokio_stream::wrappers::UnboundedReceiverStream;
+use warp::{filters::ws::Message, http::StatusCode, reply::with_header, Filter};
+
+use dotenv::dotenv;
+
+static NEXT_USER_ID: AtomicUsize = AtomicUsize::new(1);
+
+type Users = Arc<RwLock<HashMap<usize, mpsc::UnboundedSender<Result<Message, warp::Error>>>>>;
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    /// Host to bind to when using server mode
+    #[arg(short = 'n', long, default_value = "127.0.0.1")]
+    host: Option<String>,
+
+    /// Port to bind to when using server mode
+    #[arg(short, long, default_value = "8181")]
+    port: Option<String>,
+
+    /// Run local executor for testing and benchmarking
+    #[arg(short = 'x', long, default_value_t = false)]
+    executor: bool,
+
+    /// JWT token for authentication
+    #[arg(short, long)]
+    jwt: Option<String>,
+
+    /// Identity Provider (e.g., "Google")
+    #[arg(long, default_value = "IDMe")]
+    provider: Option<String>,
+}
+
+#[derive(Deserialize)]
+struct JwtRequest {
+    jwt: String,
+}
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct NopRef {
+    pub idx: u32,
+}
+
+alloy_sol_types::sol! {
+    struct ClaimsData {
+        address addr;
+        bytes ident;
+    }
+}
+
+#[derive(Deserialize)]
+struct AuthRequest {
+    code: String,
+}
+
+#[derive(Deserialize)]
+struct TokenResponse {
+    access_token: String,
+    token_type: String,
+    expires_in: u32,
+    scope: String,
+    refresh_token: String,
+    refresh_expires_in: u32,
+    id_token: String,
+}
+
+async fn auth_handler(
+    auth_request: AuthRequest,
+    users: Users,
```

Comment:
Noting that this is usused (as well as a lot of other things in this file. Shouldn't really matter since it's a demo, though.

Reply:
see note above! tldr, its a quick fork with a lot of dead code.